packages/cli/src/templates/trellis/scripts/common/task_utils.py

```
@@ -270,28 +378,73 @@ def run_task_hooks(event: str, task_json_path: Path, repo_root: Path) -> None:
     env = {**os.environ, "TASK_JSON_PATH": str(task_json_path)}
 
+    # POSIX only: a fresh session puts the shell and every descendant in one
+    # process group, which is what makes the timeout kill the whole tree.
+    popen_kwargs: dict = {}
+    if os.name == "posix":
+        popen_kwargs["start_new_session"] = True
+
     for cmd in commands:
         try:
-            result = subprocess.run(
+            with subprocess.Popen(
```

Ruff (0.16.2)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)